Assets/SEE/UI/DebugAdapterProtocol/DebugAdapterProtocolSession.cs

```diff
@@ -0,0 +1,1233 @@
+                variablesWindow.Variables = variables;
+                variablesWindow.RetrieveNestedVariables = RetrieveNestedVariables;
+                variablesWindow.RetrieveVariableValue = RetrieveVariableValue;
+                manager.AddWindow(variablesWindow);
+            }
+
+            manager.ActiveWindow = variablesWindow;
+        }
+
+
+        /// <summary>
+        /// Creates the process for the debug adapter.
+        /// </summary>
+        /// <returns>Whether the creation was sucessful.</returns>
+        private bool CreateAdapterProcess()
+        {
+            adapterProcess = new Process()
+            {
+                StartInfo = new ProcessStartInfo()
+                {
+                    FileName = Adapter.AdapterFileName,
+                    Arguments = Adapter.AdapterArguments,
+                    WorkingDirectory = Adapter.AdapterWorkingDirectory,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    RedirectStandardInput = true,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    // message headers are ASCII, message bodies are UTF-8
+                    StandardInputEncoding = Encoding.ASCII,
+                    StandardOutputEncoding = Encoding.ASCII,
+                    StandardErrorEncoding = Encoding.ASCII,
+                }
+            };
+            adapterProcess.EnableRaisingEvents = true;
+            adapterProcess.Exited += (_, args) => ConsoleWindow.AddMessage($"Process: Exited! ({(!adapterProcess.HasExited ? adapterProcess.ProcessName : null)})");
+            adapterProcess.Disposed += (_, args) => ConsoleWindow.AddMessage($"Process: Exited! ({(!adapterProcess.HasExited ? adapterProcess.ProcessName : null)})");
+            adapterProcess.OutputDataReceived += (_, args) => ConsoleWindow.AddMessage($"Process: OutputDataReceived! ({adapterProcess.ProcessName})\n\t{args.Data}");
+            adapterProcess.ErrorDataReceived += (_, args) =>
+            {
+                string message = $"Process: ErrorDataReceived! ({adapterProcess.ProcessName})\n\t{args.Data}";
+                ConsoleWindow.AddMessage(message + "\n", "Adapter", "Error");
+                Debug.LogWarning(message);
+            };
+
+            string currentDirectory = Directory.GetCurrentDirectory();
+            try
+            {
+                // working directory needs to be set manually so that executables can be found at relative paths
+                Directory.SetCurrentDirectory(Adapter.AdapterWorkingDirectory);
+                if (!adapterProcess.Start())
+                {
+                    adapterProcess = null;
+                }
+            }
+            catch (Exception e)
+            {
+                ConsoleWindow.AddMessage(e.Message + "\n", "Adapter", "Error");
+                Debug.LogWarning(e);
```

Comment:
I used `ConsoleWindow.AddMessage(..., 'Error')` in combination with `Debug.LogWarning`, 
because they just communicate that something went wrong when executing the debug session.
That doesn't mean that something went wrong in `SEE` / `Unity`.

For example that messages could be caused by:
* The selected file couldn't be found/executed.
* The debug adapter is missing.
* The executed program has thrown an error.

That aren't errors per se. (the implementation in `SEE` isn't wrong)

Reply:
I can see why, but for your given examples, I still think `LogError` would be more fitting: A logged error doesn't necessarily mean there's an implementation error in SEE, it can also be used for general error cases (e.g., code city file not found, Axivion Dashboard API not accessible, etc.). Perhaps the only one of these examples that I think would make sense as a warning is "The executed program has thrown an error", since that may be expected during debugging. The first two sound like an incorrect setup and sound like they should be errors. However, you're more familiar with the implementation here (e.g., how this is handled in other DAP clients), so we can also stick with `LogWarning` here if you think this is the better approach—I just wanted to clear up that `LogError` doesn't have to imply an error in SEE's implementation.

Reply:
If `LogError` doesn't imply that, then it's better to use that.